[Business] 주문 내역 화면 구현

jusang3057 opened the pull request:
## PR 개요
이슈 번호:

## PR 체크리스트
- [x] Code convention을 잘 지켰나요?
- [x] Lint check를 수행하였나요?
- [x] Assignees를 추가했나요?

## 작업사항
- [ ] 버그 수정
- [x] 신규 기능
- [ ] 코드 스타일 수정 (포맷팅 등)
- [ ] 리팩토링 (기능 수정 X, API 수정 X)
- [ ] 기타

### 작업사항의 상세한 설명
- 디자인/UI 레이아웃만 구성함

### 논의 사항

### 스크린샷

## 추가내용
- [x] develop, sprint 브랜치를 향하고 있습니다
- [ ] production 브랜치를 향하고 있습니다

Commit:
order history

Commit:
String 수정

jusang3057 committed:
Update build.gradle.kts

Commit:
order history compose fix + order ongoing compose

Commit:
String fix

Commit:
Merge remote-tracking branch 'origin/feature/#1012-order-history' into feature/#1012-order-history

kongwoojin commented on feature/store/src/main/java/in/koreatech/koin/feature/store/component/FilterTextChip.kt:
```diff
@@ -0,0 +1,115 @@
+import `in`.koreatech.koin.feature.store.enums.OrderFilterEnum
+
+@Composable
+fun <T> FilterTextChipSelect(
```
value를 generic으로 받아야 하는 이유가 있나요

jusang3057 replied:
Enum을 받기 위해서 이렇게 했습니다.
현재 여기 들어가야 하는 Enum이 총 4개라서..

kongwoojin replied:
parameter로 넘길 수 없는건가요

jusang3057 replied:
예 혼자 서로 다른 Enum을 4개 받아야 합니다. 그래서 interface로 한번에 받을 수 있도록 저렇게 해놨습니다(그 파일 분리해달라는 그거).
더 좋은 아이디어 있음 언제나 환영입니다. (근데 일단 저는 이게 한계인듯...)

kongwoojin replied:
TextChip이면 결국 칩에 출력할 텍스트만 필요한거 아닌가요
텍스트만 list로 넘겨도 될 것 같습니다

kongwoojin commented on feature/store/src/main/java/in/koreatech/koin/feature/store/component/FilterTextChip.kt:
```diff
@@ -0,0 +1,115 @@
+                spotColor = RebrandKoinTheme.colors.neutral500
+            )
+            .then(
+                if (isSelected) {
```
`then`을 이용해서 할 수는 없을까요

jusang3057 replied:
then으로 뭐하라는건지 모르겠습니다.

kongwoojin replied:
```
Modifier
        .clickable { onClick() }
        .then(
                if (isSelected) {
                        Modifier.border(
                            width = 1.dp,
                            color = RebrandKoinTheme.colors.neutral300,
                            shape = CircleShape
                        )
              } else {
                      Modifier
              }
        )
```

jusang3057 replied:
수정했읍니다

kongwoojin commented on feature/store/src/main/java/in/koreatech/koin/feature/store/component/FilterTextChip.kt:
```diff
@@ -0,0 +1,115 @@
+
+@Preview(showBackground = true)
+@Composable
+fun FilterTextChipSelectPreview() {
```
Preview는 private 처리 해주세요

jusang3057 replied:
수정하겠습니다.

kongwoojin commented on feature/store/src/main/java/in/koreatech/koin/feature/store/component/KoinOrdersFilterChip.kt:
```diff
@@ -0,0 +1,71 @@
+    modifier: Modifier = Modifier,
+    onClick: () -> Unit = {}
+) {
+    val backgroundColor = if (active) RebrandKoinTheme.colors.primary500 else RebrandKoinTheme.colors.neutral0
```
한 곳에서만 쓰는 것 같은데, 굳이 변수를 따로 만들 필요가 있을까요

jusang3057 replied:
수정하겠습니다.

kongwoojin commented on feature/store/src/main/java/in/koreatech/koin/feature/store/component/MenuAddButton.kt:
```diff
@@ -0,0 +1,72 @@
+    Button(
+        modifier = modifier.fillMaxWidth(),
+        onClick = onClick,
+        shape = RoundedCornerShape(8.dp),
```
cornershape 8dp는 디자인시스템에 있지 않나요

jusang3057 replied:
그러네요. 
그런데 수정하면서 찾아보니 디자인 시스템 사용 안하고 RoundedCornerShape(8.dp) 쓴것도 많던데 나중에 바꾸면 좋을까요?

kongwoojin replied:
그건 나중에 하면 될 것 같네요

kongwoojin commented on feature/store/src/main/java/in/koreatech/koin/feature/store/component/MenuAddButton.kt:
```diff
@@ -0,0 +1,72 @@
+
+@Preview(showBackground = true)
+@Composable
+fun MenuAddButtonPRE_OPENPreview() {
```
왜 함수 이름이 언더스코어가 들어가나요

jusang3057 replied:
ENUM에 따라 프리뷰 만든다고 Enum을 복사했더니 이런 실수를..

kongwoojin commented on feature/store/src/main/java/in/koreatech/koin/feature/store/component/OrderHistoryCard.kt:
```diff
@@ -0,0 +1,195 @@
+fun OrderHistoryCard(
+    orderdata: OrderHistoryData,
+    modifier: Modifier = Modifier,
+    onDetailClick: () -> Unit,
```
lambda들 default parameter 선언해주세요

jusang3057 replied:
수정하겠습니다.

kongwoojin commented on feature/store/src/main/java/in/koreatech/koin/feature/store/component/OrderHistoryCard.kt:
```diff
@@ -0,0 +1,195 @@
+            }
+
+            HorizontalDivider(
+                thickness = 1.dp,
```
기본 thinkness가 1.dp라 굳이 선언 할 필요 없어보입니다

jusang3057 replied:
새로운거 알아갑니다

kongwoojin commented on feature/store/src/main/java/in/koreatech/koin/feature/store/component/OrderHistoryCard.kt:
```diff
@@ -0,0 +1,195 @@
+                    contentDescription = "",
+                    modifier = Modifier
+                        .size(88.dp)
+                        .clip(RoundedCornerShape(4.dp)),
```
얘도 디자인시스템에 선언되어 있습니다.

kongwoojin commented on feature/store/src/main/java/in/koreatech/koin/feature/store/component/OrderHistoryCard.kt:
```diff
@@ -0,0 +1,195 @@
+                        .clip(RoundedCornerShape(4.dp)),
+                    contentScale = ContentScale.Crop
+                )
+                Spacer(modifier = Modifier.width(11.dp))
```
오... 11.dp 맞나요

jusang3057 replied:
꼼꼼하게 보시네요. 
근데 11.dp맞습니다.

어.. 그리고 디자인된거랑 여기랑 살짝 다른것도 있어요. 
디자인에 좀 이상한것도 있어서.. 어디는 왼쪽 7dp 오른쪽 9dp여백이길래 8dp로 통일한것도 있고.. 이정도는 괜찮겠죠?

kongwoojin replied:
그 정도는 알잘딱깔센 하시면 됩니다

jusang3057 replied:
쉬세요!!! 왜 거기까지 가서

kongwoojin commented on feature/store/src/main/java/in/koreatech/koin/feature/store/component/OrderOnGoingCard.kt:
```diff
@@ -0,0 +1,185 @@
+    orderdata: OrderOnGoingData,
+    modifier: Modifier = Modifier
+) {
+    val chipIcon = if (orderdata.status == TypeOption.DELIVERY) {
```
`TypeOption`이 enum인 것 같은데, when 써주세요

jusang3057 replied:
넵

kongwoojin commented on feature/store/src/main/java/in/koreatech/koin/feature/store/component/OrderOnGoingCard.kt:
```diff
@@ -0,0 +1,185 @@
+                border = BorderStroke(1.dp, RebrandKoinTheme.colors.primary500)
+            ) {
+                Text(
+                    text = "주문 상세 보기",
```
string 하드코딩하지 말아주세요

jusang3057 replied:
ㅠㅠ 본다고 다 확인했는데

kongwoojin commented on feature/store/src/main/java/in/koreatech/koin/feature/store/enums/OrderFilterOptions.kt:
```diff
@@ -0,0 +1,34 @@
+    val isActivated: Boolean
+}
+
+enum class LocationOption(@StringRes override val stringRes: Int, override val isActivated: Boolean) : OrderFilterEnum {
```
enum 분리해주세요

jusang3057 replied:
모두 OrderFilterEnum상속하는 애들이라 붙여놨는데 분리한다면 OrderFilterEnum도 따로 해서 총 5개 파일로 분리하면 되는건가요?

kongwoojin replied:
enum을 상속하는 이유가 있나요?

jusang3057 replied:
LocationOption, PeriodOption, TypeOption, StatusOption을 Enum으로 만들고 Ui에서도 사용중입니다. 그래서 Enum안에 @StringRes가 저장되어 있고 Ui에서는 이걸 가지고 출력해주는데, @StringRes가 있다는 증거(?)가 enum상속입니다. 
OrderFilterEnum는 이거 상속하면 @StringRes있음을 증명해주는 애입니다.

kongwoojin commented on feature/store/src/main/java/in/koreatech/koin/feature/store/view/orders/history/FilterLayout.kt:
```diff
@@ -0,0 +1,200 @@
+@Composable
+fun FilterOverlay(
+    backgroundContent: @Composable () -> Unit,
+    onClose: () -> Unit,
```
default parameter 선언 해주시고, modifer를 최상단으로 올려주세요

jusang3057 replied:
넵

kongwoojin commented on feature/store/src/main/java/in/koreatech/koin/feature/store/view/orders/history/FilterLayout.kt:
```diff
@@ -0,0 +1,200 @@
+        Box(
+            modifier = Modifier
+                .fillMaxSize()
+                .background(Color.Black.copy(alpha = 0.7f))
```
Color.Black 말고 neutral800 써주세요

jusang3057 replied:
넵 밑에있는것도 0으로 바꿀게요

kongwoojin commented on feature/store/src/main/java/in/koreatech/koin/feature/store/view/orders/history/FilterLayout.kt:
```diff
@@ -0,0 +1,200 @@
+                .fillMaxWidth()
+                .align(Alignment.BottomCenter)
+                .background(
+                    color = Color.White,
```
이것도요

jusang3057 replied:
아 이것도 해놓으셨네

kongwoojin commented on feature/store/src/main/java/in/koreatech/koin/feature/store/view/orders/history/FilterLayout.kt:
```diff
@@ -0,0 +1,200 @@
+                modifier = Modifier.padding(vertical = 12.dp)
+            ) {
+                Row(
+                    modifier = Modifier
```
modifier에 옵션이 하나라면 줄바꿈 하지 말아주세요

jusang3057 replied:
근데 하나가 긴거면 어케하죠? shadow같이 많은 요소들 필요한거

kongwoojin replied:
그러면 옵션 괄호 안에서 줄바꿈 처리해주세요

kongwoojin commented on feature/store/src/main/java/in/koreatech/koin/feature/store/view/orders/history/FilterLayout.kt:
```diff
@@ -0,0 +1,200 @@
+                HorizontalDivider(
+                    modifier = Modifier
+                        .padding(vertical = 16.dp),
+                    thickness = 1.dp,
```
이것도 불필요해보입니다

kongwoojin commented on feature/store/src/main/java/in/koreatech/koin/feature/store/view/orders/history/FilterLayout.kt:
```diff
@@ -0,0 +1,200 @@
+
+                    Button(
+                        modifier = Modifier.weight(1f),
+                        onClick = { },
```
버튼인데 onclick이 없나요

jusang3057 replied:
음.. 일단은 디자인만 한거라 차후 추가하겠습니다.

kongwoojin commented on feature/store/src/main/java/in/koreatech/koin/feature/store/view/orders/history/FilterLayout.kt:
```diff
@@ -0,0 +1,200 @@
+                        modifier = Modifier.weight(1f),
+                        onClick = { },
+                        shape = RoundedCornerShape(8.dp),
+                        contentPadding = PaddingValues(vertical = 10.dp, horizontal = 64.dp),
```
64dp가 맞나요

jusang3057 replied:
네 디자인상으로는 둘다 64dp인데 오른쪽게 크게 되어있더라고요.
오히려 왼쪽걸 어케만들어야 하는지 몰라서 일단 대충 비슷하게 했습니다.

kongwoojin commented on feature/store/src/main/java/in/koreatech/koin/feature/store/view/orders/history/OrderHistoryScreen.kt:
```diff
@@ -0,0 +1,195 @@
+            }
+            item {
+                KoinOrdersFilterChip(
+                    text = stringResource(filters.type.stringRes) + stringResource(R.string.bullet_separator) + stringResource(filters.status.stringRes),
```
그냥 하나의 string으로 만들면 안되나요

jusang3057 replied:
하나의 string으로 만들면 2*2해서 총 4개의 String이 필요하고 type이나 status가 추가되면 더 늘어나게 됩니다.
근데 저렇게 하면 기존에 string재활용할 수 있어서 저렇게 했어요.
저게 필터 Clip에 있는 텍스트 그대로 들어가는 거라서..

kongwoojin commented on feature/store/src/main/java/in/koreatech/koin/feature/store/view/orders/ongoing/OrderOnGoingScreen.kt:
```diff
@@ -0,0 +1,141 @@
+            ) {
+                Box {
+                    Icon(
+                        modifier = Modifier.width(95.dp),
```
icon인데 width만 고정해도 괜찮나요?

jusang3057 replied:
height는 비율에 맞춰서 자동 조절입니다.

kongwoojin commented on feature/store/src/main/java/in/koreatech/koin/feature/store/view/orders/ongoing/OrderOnGoingScreen.kt:
```diff
@@ -0,0 +1,141 @@
+            }
+        }
+    } else {
+        LazyColumn(
```
전체를 LazyColumn으로 감쌀 수는 없나요

jusang3057 replied:
결국엔 데이터가 없을때 중앙에 아이콘, 안내문구를 띄워야해서 Alignment.Center를 쓰기 위한 무언가가 필요할텐데, LazyColumn으로 전체를 감쌌을때의 이점을 모르겠습니다.

kongwoojin commented on gradle/libs.versions.toml:
```diff
@@ -185,6 +186,7 @@ strict-version-matcher-plugin = { module = "com.google.android.gms:strict-versio
 timber = { module = "com.jakewharton.timber:timber", version.ref = "timber" }
 toolbar-compose = { module = "me.onebone:toolbar-compose", version.ref = "toolbarCompose" }
 turbine = { module = "app.cash.turbine:turbine", version.ref = "turbine" }
+androidx-benchmark-traceprocessor-android = { group = "androidx.benchmark", name = "benchmark-traceprocessor-android", version.ref = "benchmarkTraceprocessorAndroid" }
```
이건 왜 추가한건가요

jusang3057 replied:
아 ALT-Enter잘못쳐서 Gradle에 들어간거 지웠는데 여기서 남아있네요

kongwoojin reviewed (changes requested):
고생하셨습니다.
코멘트 확인해주세요

Commit:
1차 리뷰 수정

kongwoojin commented on feature/store/src/main/java/in/koreatech/koin/feature/store/component/FilterTextChip.kt:
```diff
@@ -0,0 +1,114 @@
+            horizontalArrangement = Arrangement.spacedBy(12.dp)
+        ) {
+            value::class.java.enumConstants
+                .drop(1)
```
drop(1)은 왜 하는건가요

jusang3057 replied:
필터를 표시해주는 곳이 2군데 있어서 유지보수 편하려고 2개를 같은 데이터를 사용합니다.
근데 다른쪽에서는 디폴트값이 필요해서 1번째 값을 디폴트 값으로 설정했습니다.
근데 여기서는 그 첫번째 값이 필요 없어서 뺀겁니다.

jusang3057 replied:
근데 좀 가독성면에서 수정해야할것 같긴하네요

jusang3057 replied:
drop(1)대신 Enum에 isDefault라는 Boolean값 넣어서 해당 값이 아닌 값(유저선택지) 만 출력되게 하였습니다.

kongwoojin reviewed (changes requested):
고생하셨습니다.
추가로 Git convention 지켜주세요

KYM-P commented on feature/store/src/main/java/in/koreatech/koin/feature/store/component/OrderHistoryCard.kt:
```diff
@@ -0,0 +1,194 @@
+                        style = RebrandKoinTheme.typography.medium14,
+                        color = RebrandKoinTheme.colors.neutral800
+                    )
+                    Spacer(modifier = Modifier.height(6.dp))
```
상관은 없지만 verticalArrangement 로 spacedBy(6.dp) 로 줄일 수 있을 거 같습니다.

jusang3057 replied:
좋은것 같습니다.

KYM-P commented on feature/store/src/main/java/in/koreatech/koin/feature/store/component/OrderOnGoingCard.kt:
```diff
@@ -0,0 +1,178 @@
+                        style = RebrandKoinTheme.typography.medium14,
+                        color = RebrandKoinTheme.colors.neutral800
+                    )
+                    Spacer(modifier = Modifier.height(6.dp))
```
여기도 줄일 수 있을 거 같네요

jusang3057 replied:
👍

KYM-P commented on feature/store/src/main/java/in/koreatech/koin/feature/store/view/orders/history/FilterLayout.kt:
```diff
@@ -0,0 +1,197 @@
+
+            Column(
+                modifier = Modifier
+                    .padding(horizontal = 32.dp)
```
사소하지만 Modifier.padding() 으로 한줄로 줄여주세용
현재 저희 코드는 1개의 요소는 1줄로 적고있기에 맞추는게 더 좋아보입니다.

jusang3057 replied:
👍

KYM-P commented on feature/store/src/main/java/in/koreatech/koin/feature/store/view/orders/history/FilterLayout.kt:
```diff
@@ -0,0 +1,197 @@
+
+                HorizontalDivider(
+                    modifier = Modifier
+                        .padding(vertical = 16.dp),
```
HorizontalDivider 쪽에 padding 이 대부분 두줄로 적혀있네요 확인 부탁드립니다.

jusang3057 replied:
👍

KYM-P commented on feature/store/src/main/java/in/koreatech/koin/feature/store/view/orders/ongoing/OrderOnGoingScreen.kt:
```diff
@@ -0,0 +1,140 @@
+    if (orderOnGoings.isEmpty()) {
+        Box(
+            modifier = modifier
+                .fillMaxSize(),
```
한줄이 좋아보입니다.

jusang3057 replied:
👍

KYM-P reviewed (commented):
전체적으로 문제 없는 것 같습니다.
GOOD

Commit:
리뷰 수정 2

Commit:
viewmodel

jusang3057 committed:
Merge branch 'develop-business' into feature/#1012-order-history

Commit:
[add]Add API

Commit:
Merge remote-tracking branch 'origin/feature/#1012-order-history' into feature/#1012-order-history

kongwoojin commented on data/src/main/java/in/koreatech/koin/data/api/StoreApi.kt:
```diff
@@ -145,4 +147,17 @@ interface StoreApi {
     ): OrderableShopSearchRelatedResponse
+
+    @GET("order")
+    suspend fun getOrderHistories(
```
이건 AuthApi 아닌가요

kongwoojin commented on domain/src/main/java/in/koreatech/koin/domain/usecase/store/GetOnGoingRelatedUseCase.kt:
```diff
@@ -0,0 +1,11 @@
+class GetOnGoingRelatedUseCase @Inject constructor(
+    private val storeRepository: StoreRepository
+) {
+    suspend operator fun invoke(): List<OrderOnGoingRelated> = storeRepository.getOrderOnGoings()
```
Result로 반환해야 하지 않나요?

kongwoojin commented on feature/store/src/main/java/in/koreatech/koin/feature/store/component/FilterTextChip.kt:
```diff
@@ -13,22 +13,21 @@ import androidx.compose.runtime.Composable
-    var default: T = value::class.java.enumConstants.first()
+    onValueChange: (Int) -> Unit = {}
+) {
+    var defaultIdx = 0
```
defaultIdx를 선언한 이유가 있나요?
remember 처리도 안되어있네요

kongwoojin commented on feature/store/src/main/java/in/koreatech/koin/feature/store/component/OrderOnGoingCard.kt:
```diff
@@ -40,9 +41,9 @@ fun OrderOnGoingCard(
     modifier: Modifier = Modifier
 ) {
-    val (chipIcon, timeGuide) = when (orderdata.status) {
-        TypeOption.TAKEOUT -> R.drawable.ic_order_takeout to R.string.takeout_time_guide
```
`CartType`은 사용할 수 없나요?

kongwoojin commented on feature/store/src/main/java/in/koreatech/koin/feature/store/enums/OrderStatus.kt:
```diff
@@ -3,8 +3,19 @@ package `in`.koreatech.koin.feature.store.enums
+    COOKING(R.string.order_ongoing_status_cooking, R.string.order_ongoing_status_cooking_message, true),
+    PACKAGED(R.string.order_ongoing_status_packaged, R.string.order_ongoing_status_packaged_message),
+
+//    PICKED_UP(R.string.order_ongoing_status_picked_up),
```
주석 처리 할 필요가 있나요

jusang3057 replied:
아에 삭제하겠습니다.

kongwoojin commented on feature/store/src/main/java/in/koreatech/koin/feature/store/model/OrderHistoryData.kt:
```diff
@@ -1,18 +1,44 @@
+
+fun OrderHistoryOrders.toOrderHistoryData(): OrderHistoryData {
+    val formattedOrderDate = try {
+        val inputFormat = SimpleDateFormat("yyyy-MM-dd", Locale.getDefault())
```
얘는 왜 Locale이 `Locale.getDefault()`고 아래는 왜 `Locale.Korea`인가요

jusang3057 replied:
원래 Locale.getDefault()사용했는데(이건 숫자만 잇어서 getDefault든 Korea든 상관x), 아래 있는것은 한국어로 표시하기 위해 Locale.Korea했습니다.

kongwoojin commented on feature/store/src/main/java/in/koreatech/koin/feature/store/model/OrderHistoryData.kt:
```diff
@@ -1,18 +1,44 @@
+        val outputFormat = SimpleDateFormat("M월 d일 (E)", Locale.KOREA)
+        val date = inputFormat.parse(orderDate)
+        outputFormat.format(date!!)
+    } catch (e: Exception) {
```
Exception이 뜰 경우가 있나요
Exception이 뜨면 안될 것 같은데요

jusang3057 replied:
음.. null값 들어오는거 생각한건데 그냥 notnull 감지하는게 더 좋겠네요

kongwoojin commented on feature/store/src/main/java/in/koreatech/koin/feature/store/model/OrderOnGoingData.kt:
```diff
@@ -1,12 +1,42 @@
+
+fun OrderOnGoingRelated.toOrderOnGoingData(): OrderOnGoingData {
+    val formattedEstimatedAt = try {
+        val inputFormat = SimpleDateFormat("HH:mm", Locale.getDefault())
```
여기도네요

kongwoojin commented on feature/store/src/main/java/in/koreatech/koin/feature/store/view/orders/OrderHistoryViewModel.kt:
```diff
@@ -22,83 +22,59 @@ import org.orbitmvi.orbit.viewmodel.container
     private val getUserStatusUseCase: GetUserStatusUseCase
 ) : ViewModel(), ContainerHost<OrderHistoryState, OrderHistorySideEffect> {
     override val container = container<OrderHistoryState, OrderHistorySideEffect>(OrderHistoryState())
+    private var page = 0
```
이건 뭔가요
state라면 `OrderhistoryState`로 옮기는게 맞을 것 같고, 아니면 지역변수로 만들어 주세요

kongwoojin commented on feature/store/src/main/java/in/koreatech/koin/feature/store/view/orders/OrderHistoryViewModel.kt:
```diff
@@ -22,83 +22,59 @@ import org.orbitmvi.orbit.viewmodel.container
+    }
+
+    fun getNewOrderHistoryData() {
+        page = 0
```
State네요
이렇게 작성하는건 부적절해보입니다.

jusang3057 replied:
state로 옮기겠습니다.

kongwoojin commented on feature/store/src/main/java/in/koreatech/koin/feature/store/view/orders/OrderHistoryViewModel.kt:
```diff
@@ -22,83 +22,59 @@ import org.orbitmvi.orbit.viewmodel.container
+            intent {
+                getHistoryRelatedUseCase(
+                    page = page,
+                    limit = 10,
```
limit 10은 어디서 나온 값인가요

jusang3057 replied:
기본값을 사용하도록 없엤습니다.

kongwoojin commented on feature/store/src/main/java/in/koreatech/koin/feature/store/view/orders/OrderHistoryViewModel.kt:
```diff
@@ -22,83 +22,59 @@ import org.orbitmvi.orbit.viewmodel.container
+
+    fun getOrderHistoryData() {
+        if (page < totalPage) {
+            page += 1
```
page값을 왜 여기서 + 처리를 하나요
androix.paging을 사용해서 paging 처리를 하거나 ui 부분에서 state flow로 캡쳐해서 page 수 증가시키는게 맞을 것 같네요

jusang3057 replied:
+가 아닌 API에서 리턴되는 페이지로 하겠습니다.

kongwoojin commented on feature/store/src/main/java/in/koreatech/koin/feature/store/view/orders/history/OrderHistoryScreen.kt:
```diff
@@ -55,96 +62,149 @@ fun OrderHistoryScreen(
-                    modifier = Modifier.weight(1f),
-                    hint = stringResource(R.string.order_history_searchbar_hint)
-                )
+    val period: List<Pair<Int, Boolean>> = PeriodOption.values().map { it.stringRes to it.isDefault }
```
1. 이거 굳이 변수로 분리한 이유가 있을까요 inline 처리해도 될 것 같은데
2. 이거 remember 처리가 안되어있는데, 이렇게 하면 recomposition 될 때마다 다시 계산해서 성능 하락이 발생합니다

jusang3057 replied:
제거하겠습니다.

kongwoojin commented on feature/store/src/main/java/in/koreatech/koin/feature/store/view/orders/history/OrderHistoryScreen.kt:
```diff
@@ -55,96 +62,149 @@ fun OrderHistoryScreen(
+            ) {
+                Box {
+                    Icon(
+                        modifier = Modifier.width(95.dp),
```
이거 95맞나요

jusang3057 replied:
네 95입니다.
<img width="253" height="146" alt="image" src="https://github.com/user-attachments/assets/f33d9748-a432-480c-9768-b8e6719ac07f" />

jusang3057 replied:
4의 배수인 96으로 하겠습니다.

kongwoojin commented on feature/store/src/main/java/in/koreatech/koin/feature/store/view/orders/history/OrderHistoryScreen.kt:
```diff
@@ -55,96 +62,149 @@ fun OrderHistoryScreen(
+                    )
+                    Icon(
+                        modifier = Modifier
+                            .size(25.dp)
```
그냥 이런건 알잘딱깔센으로 4의 배수로 맞추시면 됩니다

jusang3057 replied:
24로 맞추겠습니다.

kongwoojin commented on feature/store/src/main/java/in/koreatech/koin/feature/store/view/orders/history/OrderHistoryScreen.kt:
```diff
@@ -55,96 +62,149 @@ fun OrderHistoryScreen(
+            snapshotFlow { listState.layoutInfo.visibleItemsInfo.lastOrNull()?.index }
+                .collect { lastVisibleIndex ->
+                    if (lastVisibleIndex == orderHistories.size - 1) {
+                        getOrderHistoryData()
```
page 값을 state에 선언하고, 여기서 page 값을 업데이트 한 다음에 sideEfffect를 사용해서 업데이트 or 마찬가지로 page값도 snapshotflow로 처리해서 업데이트 하는 방법 두가지가 있을 것 같네요

jusang3057 replied:
page값은 API에서 주는 currentPage값을 사용해서 업데이트 하려 합니다.

kongwoojin commented on feature/store/src/main/java/in/koreatech/koin/feature/store/view/orders/history/OrderHistoryScreen.kt:
```diff
@@ -55,96 +62,149 @@ fun OrderHistoryScreen(
+                modifier = Modifier.padding(horizontal = 24.dp, vertical = 16.dp),
+                verticalAlignment = Alignment.CenterVertically
+            ) {
+                if (isTyping) {
```
isTyping은 입력 중인지 여부를 나타내는 변수 아닌가요?
SearchBar가 보이는 조건은 입력 중이 아닐 것 같습니다.

jusang3057 replied:
isSearching으로 바꾸겠습니다.

kongwoojin commented on feature/store/src/main/java/in/koreatech/koin/feature/store/view/orders/history/OrderHistoryScreen.kt:
```diff
@@ -55,96 +62,149 @@ fun OrderHistoryScreen(
+                }
+                item {
+                    KoinOrdersFilterChip(
+                        text = stringResource(filters.type.stringRes) + stringResource(R.string.bullet_separator) + stringResource(filters.status.stringRes),
```
이러면 string을 쓰는 이유가 없는 것 아닌가요
string resource 하나로 수정해주세요

jusang3057 replied:
그럼 총 9가지의 경우의 수를 다 String으로 해놓나요?

kongwoojin replied:
왜 string이 9가지나 필요한지 모르겠네요
string resource를 중첩시키면 되지 않나요?

jusang3057 replied:
\<string name="bullet_separator"\>%s · %s\</string\> 이렇게 사용하라는건가요? 중첩이 이거 말씀하시는건지..

kongwoojin replied:
`stringResource(R.string.xxx, stringResource())` 이런식으로요

kongwoojin reviewed (changes requested):
고생하셨습니다.
코멘트 확인해주세요

Commit:
Merge remote-tracking branch 'origin/develop-business' into feature/#1012-order-history

Commit:
fix : in progress + history API fix

kongwoojin commented on feature/store/src/main/java/in/koreatech/koin/feature/store/model/OrderHistoryData.kt:
```diff
@@ -20,21 +20,17 @@ data class OrderHistoryData(
         orderableShopId = orderableShopId,
         orderableShopName = orderableShopName,
-        openStatus = StoreStatus.valueOf(openStatus),
+        openStatus = StoreStatus.OPEN, // openStatus,
```
API 응답값과 관계 없이 `OPEN`으로 매핑하나요?

kongwoojin commented on feature/store/src/main/java/in/koreatech/koin/feature/store/view/orders/OrderHistoryState.kt:
```diff
@@ -5,9 +5,11 @@ import `in`.koreatech.koin.feature.store.enums.StatusOption
+import `in`.koreatech.koin.feature.store.model.OrderInProgressData
 
 data class OrderHistoryState(
+    val page: Int = 0,
```
page가 현재 페이지인가요
그럼 0부터 시작하면 안될 것 같습니다

kongwoojin commented on feature/store/src/main/java/in/koreatech/koin/feature/store/view/orders/OrderHistoryViewModel.kt:
```diff
@@ -23,58 +23,61 @@ import org.orbitmvi.orbit.viewmodel.container
+        getOrderInProgressData()
     }
 
     fun getNewOrderHistoryData() {
```
무슨 역할을 하는 함수인가요

jusang3057 replied:
getOrderInProgressUseCase 실행해서 데이터 가져오는 함수입니다. OnGoing-> InProgress로 알아보기 쉽게 API와 이름을 동일하게 바꿨습니다.

kongwoojin commented on data/src/main/java/in/koreatech/koin/data/api/StoreApi.kt:
```diff
@@ -150,14 +149,11 @@ interface StoreApi {
-        @Query("status") status: String,
-        @Query("type") type: String,
-        @Query("query") query: String
+        @Query("page") page: Int?,
```
해당 query들이 nullable한 값인가요

jusang3057 replied:
네 모두 Default value가 있습니다.

kongwoojin commented on feature/store/src/main/java/in/koreatech/koin/feature/store/component/FilterTextChip.kt:
```diff
@@ -49,10 +49,8 @@ fun FilterTextChipSelect(
                         isSelected = isSelected,
                         onClick = { onValueChange(idx) },
-                        onCancle = { onValueChange(defaultIdx) }
+                        onCancle = { onValueChange(0) }
```
오타 있습니다

Commit:
fix : StringRes fix

kongwoojin commented on feature/store/src/main/java/in/koreatech/koin/feature/store/model/OrderHistoryData.kt:
```diff
@@ -20,21 +20,17 @@ data class OrderHistoryData(
-    } catch (e: Exception) {
-        orderDate
-    }
+    val inputFormat = SimpleDateFormat("yyyy-MM-dd", Locale.getDefault())
```
되도록이면 `DateTimeFormatter`를 사용해주세요
[https://javagyansite.com/2023/01/22/datetimeformatter-vs-simpledateformat/](https://javagyansite.com/2023/01/22/datetimeformatter-vs-simpledateformat/)

kongwoojin reviewed (changes requested):
고생하셨습니다.
코멘트 확인해주세요

Commit:
fix : DateFormatter change

Commit:
fix : state page default value fix

Commit:
fix : hardcoded default value fix

KYM-P commented on feature/store/src/main/java/in/koreatech/koin/feature/store/view/orders/OrderHistoryScreen.kt:
```diff
@@ -0,0 +1,214 @@
+        ) { _, _, _, _ -> }.collect {
+            viewModel.fetchData()
+        }
+    }*/
```
아직완료 안된 작업이라 빼두신 건가요?

jusang3057 replied:
삭제하겠습니다.

KYM-P commented on feature/store/src/main/java/in/koreatech/koin/feature/store/view/orders/OrderHistoryScreen.kt:
```diff
@@ -0,0 +1,214 @@
+
+    LaunchedEffect(Unit) {
+        viewModel.getUserType()
+    }
```
viewModel init 쪽에 선언하는게 더 맞는 것 같습니다.
viewModel init 이 호출 시기가 더 빠르고
이러면 한번 그려짐 > launchedEffect 로 userType 갱신 > 다시 그려짐
의 과정이라 좀 더 비효율이 아닐까요?
단순 Usecase 호출이면 NPE도 안뜰 거 같구요.

jusang3057 replied:
다른 곳에서도 해당 방식을 사용중이라 찾아봤는데 
ui와 연결된 데이터이기 때문에 ui쪽에서 실행해주는게 안전할 것 같습니다. 
오히려 제 init쪽에 있는 api도 ui에서 실행하는게 좋을것 같습니다.

KYM-P commented on feature/store/src/main/java/in/koreatech/koin/feature/store/component/SearchBarFake.kt:
```diff
@@ -52,8 +53,8 @@ fun SearchBarFake(
-                text = hint,
-                style = RebrandKoinTheme.typography.regular14.copy(color = RebrandKoinTheme.colors.neutral400)
+                text = if (query == "") hint else query,
+                style = if (query == "") RebrandKoinTheme.typography.regular14.copy(color = RebrandKoinTheme.colors.neutral400) else RebrandKoinTheme.typography.regular14.copy(color = RebrandKoinTheme.colors.neutral600)
```
이렇게 해도 상관은 없는데
query.isEmpty 가 더 직관적일 거 같습니다.

jusang3057 replied:
수정하겠습니다.

KYM-P reviewed (commented):
큰 문제는 없는 것 같습니다.

Commit:
fix : Trigger use case on ui

Commit:
fix : naming and syntax

kongwoojin commented on feature/store/src/main/java/in/koreatech/koin/feature/store/component/SearchBarFake.kt:
```diff
@@ -52,10 +52,18 @@ fun SearchBarFake(
                 tint = RebrandKoinTheme.colors.neutral500
             )
             Spacer(modifier = Modifier.width(8.dp))
-            Text(
```
`isEmpty()`인지 `== ""`인지를 제외하고 차이가 없어보이는데 수정한 이유가 있을까요

jusang3057 replied:
isEmpty()로 바꾸기 위해 수정했습니다.

kongwoojin commented on feature/store/src/main/java/in/koreatech/koin/feature/store/view/orders/OrderHistoryScreen.kt:
```diff
@@ -64,33 +65,23 @@ fun OrderHistoryScreen(
 
     BackHandler {
```
enabled parameter를 사용해도 될 것 같습니다

jusang3057 replied:
enabled parameter를 어떻게 사용하란건지 잘 모르겟습니다.

kongwoojin replied:
https://developer.android.com/develop/ui/compose/libraries?hl=ko#handling_the_system_back_button
BackHandler에는 enabled parameter가 존재합니다

jusang3057 replied:
수정했습니다.

kongwoojin commented on feature/store/src/main/java/in/koreatech/koin/feature/store/view/orders/OrderHistoryViewModel.kt:
```diff
@@ -29,52 +29,68 @@ class OrderHistoryViewModel @Inject constructor(
+    fun getNewOrderHistoryData() = intent {
+        reduce {
+            state.copy(
+                page = 1,
```
page와 totalPage값이 하드코딩 된 이유가 있나요

jusang3057 replied:
맨 처음에 불러올때라 첫 페이지 가져오기 위함입니다.

kongwoojin commented on feature/store/src/main/java/in/koreatech/koin/feature/store/view/orders/OrderHistoryViewModel.kt:
```diff
@@ -29,52 +29,68 @@ class OrderHistoryViewModel @Inject constructor(
+            }.onFailure {
+                reduce {
+                    state.copy(
+                        isLoading = true
```
failure시에도 isLoading이 false가 되어야 하지 않나요

kongwoojin commented on feature/store/src/main/java/in/koreatech/koin/feature/store/view/orders/OrderHistoryViewModel.kt:
```diff
@@ -29,52 +29,68 @@ class OrderHistoryViewModel @Inject constructor(
+                }.onFailure {
+                    reduce {
+                        state.copy(
+                            isLoading = true
```
마찬가지입니다

kongwoojin reviewed (changes requested):
고생하셨습니다. 코멘트 확인해주세요

Commit:
chore : Update Searchbar

Commit:
fix : isLoading state fix

Commit:
chore: Change file location

Commit:
test : delete Test code

Commit:
fix : BackHandler enabled fix

kongwoojin reviewed (changes requested):
`{"code":"","message":"올바르지 않은 인증정보입니다. userId is null","errorTraceId":"0055e2e6-c6a8-4649-ab82-8bfa374554db"}`
오류 발생합니다
수정해주세요

jusang3057 commented:
@kongwoojin 저도 동일한 오류가 발생한적이 있는데, 애뮬레이터에서 처음 오류난 다음 이후로 발견되지 않습니다. 오늘도 애뮬레이터에서 처음 오류난 후 앱 삭제/재설치, 앱 캐시 삭제, 앱 데이터 삭제한뒤 다시 빌드 등 다양한 방법을 사용해봤는데 오류가 재현이 안되는 상황입니다. 심지어 컴퓨터도 껐다 켜도 해당 오류가 재현이 안됩니다. 도저히 원인을 모르겠습니다.

kongwoojin commented:
> @kongwoojin 저도 동일한 오류가 발생한적이 있는데, 애뮬레이터에서 처음 오류난 다음 이후로 발견되지 않습니다. 오늘도 애뮬레이터에서 처음 오류난 후 앱 삭제/재설치, 앱 캐시 삭제, 앱 데이터 삭제한뒤 다시 빌드 등 다양한 방법을 사용해봤는데 오류가 재현이 안되는 상황입니다. 심지어 컴퓨터도 껐다 켜도 해당 오류가 재현이 안됩니다. 도저히 원인을 모르겠습니다.

해당 오류는 API 호출 시 access token을 보내지 않아서 발생하는 오류입니다.
현재 올려주신 PR에서는 **해당 오류가 나지 않을 수 없습니다.**

Commit:
fix : Api Authorization fix

kongwoojin commented on feature/store/src/main/java/in/koreatech/koin/feature/store/component/KoinOrdersFilterChip.kt:
```diff
@@ -0,0 +1,70 @@
+            .clip(RoundedCornerShape(24.dp))
+            .background(if (active) RebrandKoinTheme.colors.primary500 else RebrandKoinTheme.colors.neutral0)
+            .padding(horizontal = 8.dp, vertical = 6.dp)
+            .clickable { onClick() },
```
clickable 위치 조정해서 클릭 시 이펙트가 Chip 전체를 채우도록 하는게 좋아보입니다

kongwoojin commented on feature/store/src/main/java/in/koreatech/koin/feature/store/component/SearchBarFake.kt:
```diff
@@ -51,16 +52,32 @@ fun SearchBarFake(
 
 @Preview
 @Composable
 fun SearchBarFakePreview() {
```
private 처리 해주세요

kongwoojin commented on feature/store/src/main/java/in/koreatech/koin/feature/store/enums/OrderFilterOptions.kt:
```diff
@@ -0,0 +1,30 @@
+import androidx.annotation.StringRes
+import `in`.koreatech.koin.feature.store.R
+
+/*
```
사용하지 않는 enum 삭제해주세요

kongwoojin commented on feature/store/src/main/java/in/koreatech/koin/feature/store/model/OrderHistoryData.kt:
```diff
@@ -0,0 +1,41 @@
+)
+
+fun OrderHistoryOrders.toOrderHistoryData(): OrderHistoryData {
+    val inputFormatter = DateTimeFormatter.ofPattern("yyyy-MM-dd", Locale.getDefault())
```
서버에서 넘어오는 형태랑 다릅니다.
수정해주세요

jusang3057 replied:
서버 예시에 "2025-09-14"이렇게 되어잇는데 yyyy-MM-dd맞는것 같습니다.

kongwoojin replied:
예시가 잘못되어 있는데, 실제 API call해서 테스트는 안하신건가요

kongwoojin commented on feature/store/src/main/java/in/koreatech/koin/feature/store/model/OrderHistoryData.kt:
```diff
@@ -0,0 +1,41 @@
+
+fun OrderHistoryOrders.toOrderHistoryData(): OrderHistoryData {
+    val inputFormatter = DateTimeFormatter.ofPattern("yyyy-MM-dd", Locale.getDefault())
+    val outputFormatter = DateTimeFormatter.ofPattern("M월 d일 (E)", Locale.KOREA)
```
결국 string으로 넘기는건데, DateTime으로 넘기는게 낫지 않을까요

jusang3057 replied:
근데 DateTime으로 넘기면 reComposition될때마다 계속 fommat해야하는데요..

kongwoojin replied:
remember로 처리하면 되지 않나요

kongwoojin commented on feature/store/src/main/java/in/koreatech/koin/feature/store/model/OrderInProgressData.kt:
```diff
@@ -0,0 +1,43 @@
+)
+
+fun OrderInProgress.toOrderInProgressData(): OrderInProgressData {
+    val formattedEstimatedAt = if (estimatedAt != null) {
```
마찬가지입니다

kongwoojin commented on feature/store/src/main/java/in/koreatech/koin/feature/store/orders/OrderHistoryScreen.kt:
```diff
@@ -0,0 +1,204 @@
+        viewModel.getUserType()
+    }
+
+    if (uiState.showSignInDialog) {
```
비로그인 시 로그인 유도 다이얼로그가 표시되지 않는 것 같습니다

kongwoojin commented on feature/store/src/main/java/in/koreatech/koin/feature/store/orders/OrderHistoryScreen.kt:
```diff
@@ -0,0 +1,204 @@
+    ) {
+        Column(
+            modifier = modifier
+                .fillMaxSize()
```
줄바꿈 없애주세요

kongwoojin commented on feature/store/src/main/java/in/koreatech/koin/feature/store/orders/OrderHistoryScreen.kt:
```diff
@@ -0,0 +1,204 @@
+        modifier = modifier.fillMaxSize()
+    ) {
+        Column(
+            modifier = modifier
```
하나의 modifier을 왜 계속 사용하는건가요

kongwoojin commented on feature/store/src/main/java/in/koreatech/koin/feature/store/orders/OrderHistoryScreen.kt:
```diff
@@ -0,0 +1,204 @@
+                    OrderHistoryScreen(
+                        filters = uiState.filters,
+                        orderHistories = uiState.orderHistories,
+                        modifier = modifier,
```
여기도 마찬가지입니다

kongwoojin commented on feature/store/src/main/java/in/koreatech/koin/feature/store/orders/OrderHistoryState.kt:
```diff
@@ -0,0 +1,28 @@
+
+data class OrderHistoryState(
+    val page: Int = 1,
+    val totalPage: Int = 2,
```
placeholder라면 1로 하면 안되는건가요

jusang3057 replied:
무한 api 호출을 방지하기 위해 page 가 totalPage 보다 작을때만 요청됩니다.

kongwoojin commented on feature/store/src/main/java/in/koreatech/koin/feature/store/orders/OrderHistoryViewModel.kt:
```diff
@@ -0,0 +1,237 @@
+            )
+        }
+
+        if (state.page < state.totalPage) {
```
여기 때문이군요
근데 그러면 받아온 `page`와 `totalPage`가 동일할 경우, 다시 로딩이 되지 않는거 아닌가요

jusang3057 replied:
page와  totalPage가 동일하다면 모든 정보를 불러왔다는 것이니 로딩되지 안ㄴ는게 맞는것 같습니다.
이게 없으면 맨 마지막에 내려가면 계속 api를 호출하게 됩니다.

kongwoojin commented on feature/store/src/main/java/in/koreatech/koin/feature/store/orders/OrderHistoryViewModel.kt:
```diff
@@ -0,0 +1,237 @@
+    fun getOrderHistoryData() = intent {
+        reduce {
+            state.copy(
+                isLoading = true
```
여기서 loading 전에 `isLoading`을 `true`로 바꿀거라면, 굳이 기본값이 `true`일 필요가 없어보이네요

jusang3057 replied:
근데 굳이 false로 놓을 필요도 없지 않을까요? 결국 api호출이 완료되면 false로 바뀔거니..

kongwoojin replied:
UI에 처음부터 Progress가 표시되는건 아니지 않나요
그러면 false가 맞을 것 같습니다

kongwoojin commented on feature/store/src/main/java/in/koreatech/koin/feature/store/orders/OrderHistoryViewModel.kt:
```diff
@@ -0,0 +1,237 @@
+        }
+    }
+
+    fun navigateToCart() = intent {
```
왜 ui 작업인 코드가 viewmodel에 있나요

jusang3057 replied:
?? 이건 ViewModel이 맞는것 같은데요..

kongwoojin replied:
UI 쪽 작업이잖아요
그러면 ViewModel에서 처리하면 안되는게 맞지 않나요

jusang3057 replied:
어디가 ui쪽 작업이라는건지 모르겠습니다. 페이지 이동과 state관리도 ViewModel이 하는게 맞지 않나요?

kongwoojin replied:
현재 페이지가 어디인지 저장을 한다면 그건 viewmodel이겠지만, 여기는 페이지 이동 아닌가요

jusang3057 replied:
viewmodel이 직접 페이지 이동하는것도 아니고 OrderSideEffect로 이벤트 전달하는거 아닌가요? 그럼 ViewModel에서 지금 사용이 맞다고 생각합니다.

kongwoojin commented on feature/store/src/main/java/in/koreatech/koin/feature/store/orders/OrderHistoryViewModel.kt:
```diff
@@ -0,0 +1,237 @@
+        }
+    }
+
+    fun applyFilterOverlay(newFilters: OrderFilter) {
```
필터를 적용하는 함수 같은데, 이름에 overlay가 들어가면 안될 것 같습니다

kongwoojin commented on feature/store/src/main/java/in/koreatech/koin/feature/store/orders/component/FilterOverlay.kt:
```diff
@@ -0,0 +1,218 @@
+    onClose: () -> Unit = {},
+    onApply: (OrderFilter) -> Unit = {}
+) {
+    var localFilters by remember { mutableStateOf(filters) }
```
state hoisting 해주세요

jusang3057 replied:
localFilters 는 이 composable에서만 쓰이는 임시 상태인데 hoisting해도 되나요?

kongwoojin replied:
왜 굳이 filter를 Composable 안에서 처리하는건가요?

jusang3057 replied:
여기 있는 filter는 저장해야지 적용되는 것이고, 뒤로가기나 x버튼을 통해 저장안하고 나갈수 있습니다.
그래서 임시 저장용으로 만든게 localFilters이고 여기서만 사용됩니다.

kongwoojin commented on feature/store/src/main/java/in/koreatech/koin/feature/store/orders/component/FilterTextChip.kt:
```diff
@@ -0,0 +1,97 @@
+    title: String,
+    selected: Int,
+    defaultIdx: Int,
+    chipTitle: List<Pair<Int, Boolean>>,
```
이름이 title인데 왜 List<Pair>로 받는건가요

jusang3057 replied:
list<int>로 수정하였습니다.

kongwoojin replied:
item이 더 적합하지 않을까요

jusang3057 replied:
item이 뭘 말씀하시는건지 모르겠습니다.

kongwoojin replied:
`chipItem`이요

jusang3057 replied:
아 변수명 말씀하시는 거였군요 변경하겠스비다.

kongwoojin commented on feature/store/src/main/java/in/koreatech/koin/feature/store/orders/component/MenuAddButton.kt:
```diff
@@ -0,0 +1,71 @@
+        when (status) {
+            StoreStatus.PRE_OPEN -> stringResource(R.string.add_same_menu_pre_open)
+            StoreStatus.SOLD_OUT -> stringResource(R.string.add_same_menu_sold_out)
+            else -> stringResource(R.string.add_same_menu_open)
```
else 말고 `StoreStatus.OPEN`으로 처리해주세요

kongwoojin commented on feature/store/src/main/java/in/koreatech/koin/feature/store/orders/component/OrderHistoryCard.kt:
```diff
@@ -0,0 +1,203 @@
+
+    Card(
+        modifier = modifier.fillMaxWidth(),
+        shape = RoundedCornerShape(12.dp),
```
12.dp면 designsystem에 있지 않나요

kongwoojin commented on feature/store/src/main/java/in/koreatech/koin/feature/store/orders/component/OrderInProgressCard.kt:
```diff
@@ -0,0 +1,194 @@
+) {
+    val chipIcon = when (orderdata.orderType) {
+        TypeOption.TAKE_OUT -> R.drawable.ic_order_takeout
+        else -> R.drawable.ic_order_delivery
```
마찬가지입니다. `TypeOption.DELIVERY`로 처리해주세요

kongwoojin commented on feature/store/src/main/java/in/koreatech/koin/feature/store/orders/component/OrderInProgressCard.kt:
```diff
@@ -0,0 +1,194 @@
+
+            OutlinedButton(
+                modifier = Modifier.fillMaxWidth(),
+                onClick = { },
```
이건 왜 onClick이 없나요

jusang3057 replied:
이거 클릭해서 이동하는 페이지가 아직 없습니다.

kongwoojin commented on feature/store/src/main/java/in/koreatech/koin/feature/store/orders/component/OrdersTabRow.kt:
```diff
@@ -0,0 +1,80 @@
+    },
+    divider: @Composable () -> Unit = @Composable {
+        HorizontalDivider(
+            color = KoinTheme.colors.neutral400
```
RebrandKoinTheme 사용해주세요

kongwoojin commented on feature/store/src/main/java/in/koreatech/koin/feature/store/orders/screen/OrderHistoryScreen.kt:
```diff
@@ -0,0 +1,287 @@
+import `in`.koreatech.koin.feature.store.orders.component.SearchBar
+
+@Composable
+fun OrderHistoryScreen(
```
OrderHistoryScreen이 두개 존재하는데, 무슨 차이인가요

jusang3057 replied:
상위 페이지를 OrderScreen으로 변경하였습니다.

kongwoojin commented on feature/store/src/main/java/in/koreatech/koin/feature/store/orders/OrderHistoryViewModel.kt:
```diff
@@ -0,0 +1,237 @@
+        }
+    }
+
+    private fun getOrderInProgressData() {
```
얘는 왜 호출을 안하고 있나요

kongwoojin commented on feature/store/src/main/java/in/koreatech/koin/feature/store/orders/screen/OrderHistoryScreen.kt:
```diff
@@ -0,0 +1,287 @@
+            type = TypeOption.NONE,
+            status = StatusOption.NONE
+        ),
+        orderHistories = orderHistories
```
그냥 바로 listOf()를 사용하면 되지 않나요

kongwoojin reviewed (changes requested):
고생하셨습니다.
추가로 비로그인 상태에서 주문 내역 화면에 접근할 때 처리가 필요할 것 같습니다.

kongwoojin commented:
그리고 왜 `develop, sprint 브랜치를 향하고 있습니다`와 `production 브랜치를 향하고 있습니다`가 모두 체크되어있는건가요

Commit:
fix : bug fix

 - chip clickable fix
 - preview private
 - del unused enum
 - edit date format
 - show login dialog when not logined
 - ui state fix
 - file name change
 - other

Commit:
chore : change list name

Commit:
chore : change list name

Commit:
fix : Critical typo fix

Commit:
fix : activate in-progress api function call

Commit:
fix : Time datatype fix string->localtime

kongwoojin commented on gradle/libs.versions.toml:
```diff
@@ -72,6 +72,7 @@ strictVersionMatcherPlugin = "1.2.4"
 timber = "5.0.1"
 toolbarCompose = "2.3.2"
 turbine = "1.2.0"
+benchmarkTraceprocessorAndroid = "1.4.0"
```
얘는 뭔가요

kongwoojin reviewed (changes requested):
고생하셨습니다.
근데 directory structure가 정의한 것이랑 다르네요

Commit:
fix : directory structure fix

kongwoojin reviewed (approved):
한 파일에 다 때려박는 구조는 안 좋아보입니다.
그래도 코드상의 문제는 없으니 approve 하겠습니다

Commit:
chore : ktlint fix

jusang3057 commented:
@kongwoojin 밑에 바텀 내비 때문에 2 screen을 한 파일에 만들어야 하는데 규칙 지키면서 만들 방법이 생각나지 않습니다..

kongwoojin commented:
뭐 그건 심각한건 아니니 나중에 논의 해보면 될 것 같고요, lint 수정해주세요

kongwoojin commented:
수정 다 하셨으면 머지하시면 됩니다
